File: MyCustomDialogJava/app/src/main/java/io/github/d2fault/mycustomdialogjava/MainActivity.java
package io.github.d2fault.mycustomdialogjava;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button showDialogButton = findViewById(R.id.showDialogButton);
        showDialogButton.setOnClickListener(v -> {
            CustomDialogMaker.getDialog(this, "title", "message", "submit", "cancel", new CustomDialogListener() {
                @Override
                public void onClickSubmitButton() {
                    Toast.makeText(getApplicationContext(), "SUBMIT!", Toast.LENGTH_SHORT).show();
                }

                @Override
                public void onClickCancelButton() {
                    Toast.makeText(getApplicationContext(), "CANCEL!", Toast.LENGTH_SHORT).show();
                }
            });
        });
    }
}

File: MyCustomDialog/app/src/main/java/io/github/d2fault/mycustomdialog/MainActivity.kt
package io.github.d2fault.mycustomdialog

import android.os.Bundle
import android.widget.Button
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import io.github.d2fault.mycustomdialog.dialog.CustomDialogListener
import io.github.d2fault.mycustomdialog.dialog.CustomDialogMaker

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val showDialogButton = findViewById<Button>(R.id.showDialogButton)

        showDialogButton.setOnClickListener {
            CustomDialogMaker.getDialog(
                this,
                "title",
                "message",
                "submit",
                "cancel",
                object : CustomDialogListener {
                    override fun onClickSubmitButton() {
                        Toast.makeText(applicationContext, "SUBMIT!", Toast.LENGTH_SHORT).show()
                    }

                    override fun onClickCancelButton() {
                        Toast.makeText(applicationContext, "CANCEL!", Toast.LENGTH_SHORT).show()
                    }
                })
        }
    }
}